source/common/http/http1/conn_pool.cc

```diff
@@ -220,6 +229,16 @@ void ConnPoolImpl::onResponseComplete(ActiveClient& client) {
 
 void ConnPoolImpl::onUpstreamReady() {
   upstream_ready_enabled_ = false;
+  auto it = delayed_clients_.begin();
+  while (it != delayed_clients_.end()) {
+    ActiveClient& client = **it;
+    it++; // Move forward before moveBetweenLists which would invalidate 'it'.
+    client.delayed_--;
+    if (client.delayed_ == 0) {
+      ENVOY_CONN_LOG(debug, "moving from delay to ready", *client.codec_client_);
+      client.moveBetweenLists(delayed_clients_, ready_clients_);
+    }
```

Comment:
could you create another time here when 
1. the delayed_clients is not empty, and
2. there is no pending timer
So that the `delayed clients` can be moved to `ready` in next batch.

Reply:
Done.